lib/apiservers/service/restapi/handlers/common.go

```diff
@@ -33,26 +33,38 @@ import (
 	"github.com/vmware/vic/pkg/vsphere/vm"
 )
 
-func buildData(ctx context.Context, url url.URL, user string, pass string, thumbprint *string, datacenter *string, computeResource *string) (*data.Data, error) {
+type buildDataParams struct {
+	target          string
+	thumbprint      *string
+	datacenter      *string
+	computeResource *string
+}
+
+func buildData(ctx context.Context, params buildDataParams, principal interface{}) (*data.Data, error) {
 	d := data.Data{
 		Target: &common.Target{
-			URL:      &url,
-			User:     user,
-			Password: &pass,
+			URL: &url.URL{Host: params.target},
 		},
 	}
 
-	if thumbprint != nil {
-		d.Thumbprint = *thumbprint
+	if c, ok := principal.(Credentials); ok {
+		d.Target.User = c.user
+		d.Target.Password = &c.pass
+	} else {
+		d.Target.CloneTicket = principal.(Session).ticket
+	}
+
+	if params.thumbprint != nil {
```

Comment:
what do we do if there is no thumbprint?

Reply:
Being able to call the API without passing a thumbprint just requires that the vSphere target has a well-signed, or otherwise trusted, certificate (like calling the CLI without a thumbprint).